pkg/util/kubernetes/kubelet/containers.go

```diff
@@ -113,6 +115,43 @@ func parseContainerInPod(status ContainerStatus, pod *Pod) (*containers.Containe
 	return c, nil
 }
 
+func parseContainerNetworkAdresses(status ContainerStatus, pod *Pod) []containers.NetworkAddress {
+	addrList := []containers.NetworkAddress{}
+	podIP := net.ParseIP(pod.Status.PodIP)
+	if podIP == nil {
+		log.Warnf("Unable to parse pod IP: %v for pod: %s", pod.Status.PodIP, pod.Metadata.Name)
+		return addrList
+	}
+	hostIP := net.ParseIP(pod.Status.HostIP)
+	if hostIP == nil {
+		log.Warnf("Unable to parse host IP: %v for pod: %s", pod.Status.HostIP, pod.Metadata.Name)
+		return addrList
+	}
+	// Look for the ports in container spec
+	for _, s := range pod.Spec.Containers {
+		if s.Name == status.Name {
```

Comment:
nit: Maybe I'm wrong, but should we break the for loop if the correct container is found? Otherwise we will keep looping for every container every time. I know the counts are likely small but still......

Reply:
Good catch forgot to add the `break`